### PR TITLE
docs: update task leader to explain shutdown sequence.

### LINKED ISSUE #9986: [docs] updated documentation of `task->leader` in light of new lifecycle capabilities
### Issue
The docs for `task -> leader`([here](https://www.nomadproject.io/docs/job-specification/task#leader)) predate task lifecycle capabilities. These should be checked for correctness and then updated to be explicit about the semantics of `leader` in each stage and to make distinction between leader and lifecycle.

### PR DESCRIPTION
**Pull request by DerekStrickland**
Closes #9986

This PR expands the documentation for the `task.leader` field to explain the impact on shutdown processing. The issue suggests providing an explanation of `lifecycle`, but I've opted to link out to existing documentation rather than duplicating the text in multiple places.

### CODE REVIEW

**Review by lgfa29 — APPROVED** [review]
Minor question about the hooks, but LGTM. Clarifying the order tasks are shutdown is great!

File: website/content/docs/job-specification/task.mdx [hunk 1]
```diff
@@ -66,7 +66,12 @@ job "docs" {
 
 - `leader` `(bool: false)` - Specifies whether the task is the leader task of
   the task group. If set to true, when the leader task completes, all other
-  tasks within the task group will be gracefully shutdown.
+  tasks within the task group will be gracefully shutdown. The shutdown
+  process starts by firing any configured pre-kill hooks, then stops the
+  the leader task first, followed by non-sidecar and non-poststop tasks,
+  and finally sidecar tasks. Once this process completes, post-stop hooks
```
[the remark this answers is not in the document]
Comment (DerekStrickland) [reply website/content/docs/job-specification/task.mdx:72]: Great callout. The pre-kill is basically the hook to apply a shutdown delay if configured. The post-stop is a user-defined lifecycle task. I'll take another pass at clarifying this. This language is still not quite clear enough.

[the remark this answers is not in the document]
Comment (DerekStrickland) [reply website/content/docs/job-specification/task.mdx]: Updated. Mind taking another look?

**Review by lgfa29 — APPROVED** [review]

File: website/content/docs/job-specification/task.mdx (hunk 1 above)
[the remark this answers is not in the document]
Comment (lgfa29) [reply website/content/docs/job-specification/task.mdx]: Looks great! Double ✅

**Pull request merged**